### PR TITLE
[184] made sure compiles & runs with java11

### PR DESCRIPTION
**Pull request by adietish**
Signed-off-by: Andre Dietisheim <adietish@redhat.com>

### CODE REVIEW

**Comment by jeffmaury** [comment]
Would be fine if CI builds run both on Java8 and Java11 so worth updating Jenkinsfile as well

**Comment by jeffmaury** [comment]
Yes because it's probably using a default JDK. So this should be added in order to control the JDK

**Comment by adietish** [comment]
with the latest updates in this PR things are now compiling and run, but a test is failing:
```
Failures:
  LaunchingJarTest.testLaunchingSupportLibraryDetector:67
```

In this test the following fails:
```
		LibraryInfo info = util.runLaunchingSupportLibraryDetector(exe, launchingJar);
		assertTrue(info.getBootpath().length > 0);
```
The Bootpath is empty. 

This test relies on the jar at ```org.jboss.tools.rsp.launching.java/src/main/resources/launchingsupport.jar``` which I suspected of being outdated. I thus replaced it with the latest version that in Eclipse 2018-12-RC1 (```org.eclipse.jdt.launching_3.12.0.v20181119-1621.jar```) but the test still fails.

I also found another problem here: LaunchingSupportUtils copies this jar to ```~/.org.jboss.tools.rsp.data/libs/launchingsupport.jar```. I am wondering what happens if you upgrade your rsp. It wont use the latest version of the jar, it will keep using the version that it copied to your home directory. See here: https://github.com/redhat-developer/rsp-server/blob/master/framework/bundles/org.jboss.tools.rsp.launching.java/src/main/java/org/jboss/tools/rsp/internal/launching/java/util/LaunchingSupportUtils.java#L98
I think that we should either not copy the file or check if it's outdated (ex. md5, timestamp, etc.).

**Comment by adietish** [comment]
I looked into (LibraryDetector)[https://github.com/eclipse/eclipse.jdt.debug/blob/master/org.eclipse.jdt.launching/support/org/eclipse/jdt/internal/launching/support/LibraryDetector.java] - that's located within ```launchingsupport.jar``` - which is the class that fails in the test. It turns out that this class is outdated, it prints the following system properties (for a sun/openjdk vm):
* any system property that ends with ```.boot.class.path``` or if not present, ```sun.boot.class.path``` ("java.class.path" in jdk >= 9)
* ```java.ext.dirs``` (obsolete jdk > 9)
* ```java.endorsed.dirs``` (obsolete jdk >= 9)

**Comment by adietish** [comment]
@jeffmaury I can patch the LibraryDetector for the ```java.class.path``` so that it works for jdk >= 9 (the other properties dont exist any more in jdk >= 9) and then build a new jar. But given that the jar is taken as-is from Eclipse, I'm wondering if we should do that - maybe just as a temporary measure. We should rather file a bug in Eclipse and submit a patch for it. There's no bug filed in Eclipse for this currently. 
The other question is what this is used for. If this was a problem I bet users would have filed Eclipse bugs. We could also simply remove the test. 
Thoughts?

**Comment by adietish** [comment]
@jeffmaury like discussed I ignored the test that fails in java11: https://github.com/redhat-developer/rsp-server/pull/194/files#diff-0426fe49a603c8a899f9ee09880aa208R55
Please review.
Updating the jenkinsfile to run tests in jdk8 and jdk11 is here: #195

**Comment by adietish** [comment]
testPR

**Comment by adietish** [comment]
@jeffmaury ok to merge this?

File: framework/tests/org.jboss.tools.rsp.launching.java.test/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -17,12 +14,17 @@ Import-Package: com.google.gson,
...
+ javax.xml.parsers,
+ javax.annotation
 Require-Bundle: org.jboss.tools.rsp.api;bundle-version="0.0.9",
  org.jboss.tools.rsp.launching;bundle-version="0.0.9",
  org.jboss.tools.rsp.launching.java;bundle-version="0.0.9",
  org.junit,
  org.jboss.tools.locus.mockito,
- org.hamcrest.core
+ org.hamcrest.core,
+ javax.annotation;bundle-version="1.2.0"
```
Comment (jeffmaury) [inline framework/tests/org.jboss.tools.rsp.launching.java.test/META-INF/MANIFEST.MF]: If you import the bundle why importing the package

**Comment by adietish** [comment]
@jeffmaury corrected, please re-review

**Pull request merged**